packages/idyll-ast/test/ast-test.js

```diff
@@ -0,0 +1,269 @@
+    const ast = astTestVar();
+    expect(util.appendChild(ast, createComponent())).to.eql(
+      updatedASTWithNode()
+    );
+  });
+
+  it('should append multiple child nodes', function() {
+    const ast = astTestVar();
+    expect(
+      util.appendChildren(ast, [createComponent(), createComponent()])
+    ).to.eql(updatedASTWithNodes());
+  });
+
+  it('should return original ast when empty array is appended', function() {
+    const ast = astTestVar();
+    expect(util.appendChildren(ast, [])).to.eql(ast);
+  });
+
+  it('should create component without props and children', function() {
+    expect(util.createNode('testNode', 'component')).to.eql({
+      type: 'component',
+      name: 'testNode'
+    });
+  });
+
+  it('should create component with children', function() {
+    const test = util.createNode('p', 'component', null, [
+      {
+        type: 'textnode',
+        value: 'This is a new node'
+      }
+    ]);
+    expect(test).to.eql(createComponent(11));
+  });
+
+  it('should create component node with properties and children', function() {
+    expect(
+      util.createComponentNode(
+        'a',
+        {
+          href: {
+            type: 'value',
+            value: 'www.example.com'
+          }
+        },
+        [
+          {
+            type: 'textnode',
+            value: 'Link to a website'
+          }
+        ]
+      )
+    ).to.eql(createAnchorNode());
+  });
+
+  it('should create a textnode', function() {
+    expect(util.createTextNode('Link to a website')).to.eql({
+      type: 'textnode',
+      value: 'Link to a website'
+    });
+  });
+
+  it('should get children of a node', function() {
+    expect(util.getChildren(updatedASTWithNode())).to.eql(testChildren());
+  });
+
+  it('should set children for a node', function() {
+    expect(
+      util.setChildren(
+        {
+          type: 'component',
+          name: 'root'
+        },
+        testChildren()
+      )
+    ).to.eql(updatedASTWithNode());
+  });
+
+  it('should get no children for a textnode', function() {
+    expect(util.getChildren(createExampleTextNode())).to.eql([]);
+  });
+
+  it('should get no children for a var', function() {
+    expect(util.getChildren(createExampleVarNode())).to.eql([]);
+  });
+
+  it('should get no children for a data', function() {
+    expect(util.getChildren(createExampleData())).to.eql([]);
+  });
+
+  // it('should set no children for a textnode', function() {
+  //   expect(util.setChildren(createExampleTextNode(), testChildren())).to.eql(
+  //     createExampleTextNode()
+  //   );
+  // });
+
+  // it('should set no children for a var', function() {
+  //   expect(util.setChildren(createExampleVarNode(), testChildren())).to.eql(
+  //     createExampleVarNode()
+  //   );
+  // });
+
+  // it('should set no children for a data', function() {
+  //   expect(util.setChildren(createExampleData(0), testChildren())).to.eql(
+  //     createExampleData()
+  //   );
+  // });
```

Comment:
Should these commented out tests be removed or re-enabled?

Reply:
I can re-enable or swap out for similar. There are also methods that lack tests, and most methods lack comprehensive tests. I plan to improve test coverage as a next step.